app/services/proxy_load_balancer.rb

```diff
@@ -0,0 +1,28 @@
+class ProxyLoadBalancer
+  def initialize
+    all_proxies = SmartProxy.unscoped.with_features(RemoteExecutionProvider.providers.keys)
+    @jobs = get_counts(all_proxies)
+  end
+
+  # Get the least loaded proxy from the given list of proxies
+  def next(proxies)
+    ids = proxies.map(&:id)
+    proxy = @jobs.reject { |id, _| !ids.include?(id) }.min_by { |_, job_count| job_count }.first
```

Comment:
`select { |proxy, _| proxies.include?(proxy) }` would be bit more natural (both using the proxy as key and using select instead of reject here)